ruby/lib/minitest/queue.rb

```diff
@@ -168,6 +181,163 @@ def flaky?
       end
     end
 
+    class PipeQueue
+      def initialize(deliminiter: nil)
+        @in, @out = IO.pipe
+        @deliminiter = deliminiter
+      end
+
+      def push(object)
+        payload = Marshal.dump(object)
+        out.write("#{payload}#{deliminiter}")
+        payload.length
```

Comment:
Always use `String#bytesize` for this sorts of thing. In this case the strings should be BINARY, so it shouldn't change anything, but if the strings were to be UTF-8, length returns the numbers of characters which can be less that the actual string number of bytes.

So for clarity `bytesize` is way better (and also sligthly faster).